### ISSUE TITLE
WebGL canvas does not work in Internet Explorer 11

### ISSUE THREAD

**Issue by maikmerten** [mbebenita/Broadway#31]
With IE11, the demo on http://bkw.github.io/Broadway/storyDemo.html will only show a green picture:

![ie11-webgl-canvas](https://f.cloud.github.com/assets/2611432/2324404/d47b9c1c-a3c8-11e3-805d-dc5d587a8437.png)

This can be attributed to IE's incomplete WebGL support. The error console will show many instances of errors such as

> WEBGL11048: INVALID_ENUM: texImage2D: Invalid Format
> File: canvas.js, Row: 114, Column: 5

> WEBGL11094: INVALID_OPERATION: texImage2D: The enumeration LUMINANCE is currently not supported
> File: canvas.js, Row: 114, Column: 5

Although IE11 may not be a primary target for Broadway, it would still be a nice-to-have. Apparently the decoder itself is functional and performing well.

**Comment by soliton4** [comment]
@maikmerten here you go
https://github.com/mbebenita/Broadway/pull/44

**Comment by cyberork** [comment]
latest update of IE11 does fix it for me (KB3021952). a nonwebgl option is nice but it puts a significant additional load to the CPU due to the YUV -> RGB color conversion and the often slow putImaga Data call, if webgl is supported through a GPU (no emulation) the color conversion is much faster. 
The nonwebgl option is interesting for older browsers, which also typically run on older, in general slower PCs/laptops. So the nonwebgl option is only a good option for faster/mostly newer PCs that run browsers without webgl support unless you reduce framerate and load elsewhere.

**Comment by soliton4** [comment]
https://github.com/mbebenita/Broadway/pull/44

there is a asm.js fallback implementation

i will merge it into the master branch soon but you can use it as a template if you want

**Comment by soliton4** [comment]
@cyberork so it was a ie issue, not a broadway bug?

**Comment by cyberork** [comment]
I tested broadway with different IE versions, IE10 and an early IE11
version couldn't run it but the latest version of IE11 ran it correctly
without issues. I remember reading in the release notes of the latest IE 11
update about changes that suggested complete webGL support.

I did not dig into why earlier versions couldn't run broadway, so there
still might be some bugs in broadway that prevent it to run on older IE
versions but I can confirm that broadway runs on the latest IE 11 update on
windows 7 and windows 8.


Sent from my iPhone

On Mar 28, 2015, at 13:45, soliton4 <notifications@github.com> wrote:

@cyberork <https://github.com/cyberork> so it was a ie issue, not a
broadway bug?

—
Reply to this email directly or view it on GitHub
<https://github.com/mbebenita/Broadway/issues/31#issuecomment-87299652>.
